client/my-sites/checkout/composite-checkout/composite-checkout.tsx

```diff
@@ -233,6 +233,14 @@ export default function CompositeCheckout( {
 		isInitialCartLoading,
 	} );
 
+	// Remove the coupon when it doesn't exist on the server-side
+	const couponSavingsTotal = responseCart?.coupon_savings_total_integer;
+	useEffect( () => {
```

Comment:
Hm... I still think this is covering over a problem that's more fundamental. If the coupon no longer applies to the updated shopping cart, I'd expect the shopping cart endpoint to remove the coupon (or at least treat it as invalid and report a warning). I'll take a look at the bug and see if I can understand what's going on.

Reply:
I reproduced this issue and the cart response actually _does_ have the coupon removed. The bug is that `getUpdatedCouponStatus` does not allow a coupon to stop being applied even if the cart says it is:

https://github.com/Automattic/wp-calypso/blob/bafff16e004276e9de18ca7981d25ed670d84e4a/packages/shopping-cart/src/use-shopping-cart-reducer.ts#L263

Reply:
Thanks @sirbrillig! The `getUpdatedCouponStatus` now returns `invalid` when the applied coupon is invalid for the current product. Even better, the code change is more compact! 👍